generic3g/tests/Test_VariableSpec_private.pf

```diff
@@ -0,0 +1,81 @@
+#include "MAPL_TestErr.h"
+module Test_VariableSpec_private
+   use mapl3g_VariableSpec_private
+   use esmf
+   use pfunit
+   use ESMF_TestMethod_mod
+   use mapl3g_EsmfRegridder, only: EsmfRegridderParam
+   implicit none(type,external)
+
+   character(len=*), parameter :: NONZERO = 'Non-zero status returned'
+   character(len=*), parameter :: EXCEPT_MSG = 'Invalid value'
+
+contains
+
+   @Test
+   subroutine test_validate_short_name()
+      integer :: status
+      
+      call validate_short_name('F00', rc=status)
+      @assertTrue(status == 0, NONZERO)
+
+      call validate_short_name('0F00', rc=status)
+      @assertExceptionRaised()
+
+      call validate_short_name('_F00', rc=status)
+      @assertExceptionRaised()
+
+      call validate_short_name('F_', rc=status)
+      @assertTrue(status == 0, NONZERO)
+
+   end subroutine test_validate_short_name
+
+   @Test
+   subroutine test_validate_state_intent()
+      integer :: status
+
+      call validate_state_intent(ESMF_STATEINTENT_IMPORT, rc=status)
+      @assertTrue(status == 0, NONZERO)
+      
+      call validate_state_intent(ESMF_STATEINTENT_EXPORT, rc=status)
+      @assertTrue(status == 0, NONZERO)
+      
+      call validate_state_intent(ESMF_STATEINTENT_INTERNAL, rc=status)
+      @assertTrue(status == 0, NONZERO)
+
+      call validate_state_intent(ESMF_STATEINTENT_UNSPECIFIED, rc=status)
+      @assertExceptionRaised()
+      
+   end subroutine test_validate_state_intent
+
+   @Test
+   subroutine test_validate_state_item()
+      integer :: status
+      
+      call validate_state_item(ESMF_STATEITEM_STATE, rc=status)
+      @assertExceptionRaised()
```

Comment:
`STATE` **is** a valid StateItem flag.

Reply:
Removed validation of StateItem, since all values are allowed.